src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -148,6 +151,17 @@
       + "Elasticsearch rejects due to some malformation of the document itself, such as an index"
       + " mapping conflict or a field name containing illegal characters. Valid options are "
       + "'ignore', 'warn', and 'fail'.";
+  public static final String WRITE_METHOD_CONFIG = "write.method";
+  private static final String WRITE_METHOD_DOC = "What method to use for storing data to "
```

Comment:
Regarding 3 actions - I do not now, but I can image that someone have a separate pipeline to insert documents and separate one for update. In such case, upsert is not acceptable.
Actually, it is very ease to implement so I did.

Reply:
Regarding 2 configuration properties.
Not bad idea, but I do not see any point. We will have one extra conf. prop. (_doc_as_upsert_) which applies only for one _action_. 

Regarding _action_ - I find it confusing. Introducing new property with possible value `delete`. Delete is already implemented <=> if value is null.
Moreover, if we define one action, we would need actually 4 connect jobs - one for each action. At least that's how I understand it.

Reply:
Hi,
  for our way of working, I agree ```delete``` does not makes much of sense. But at less the naming I find it more clear than _write.method_, with my proposal sort of we mimic the simply model of http, sort of easy to understand actions (something like _insert_, _update_) then the boolean value will change of of this action to do both. What do you think?

speaking about use cases, I sort of see more from personal experience either:

* full insert pipeline, where every new event is a new document.
* upsert pipeline, updates and insert mixed in the pipeline.

one example of this would be for example when doing CDC with something like http://maxwells-daemon.io

At the end, my question here goes into finding a proper interface for the user that sticks and helps understanding what to do... hope it makes sense.

Reply:
So, we have several possible actions connect task can do with record while interacting with ES:
- *create*
- **
- **
- **
- **
- **

Reply:
@jdurani sorry, I don't understand your last message?

Reply:
That one was WIP :).

Besides my last comment, I am open to renaming config property.

Reply:
@purbon What do you thing about my last commet?

Reply:
thanks a lot for your kind description and feedback, I'm actually on the fence for this, option two keeps the actions to basic, then asking an action to do upsert, while the option 1 is more opinionated as well as more closed to what users might end up doing anyway.

I guess is ok for now to have simple actions:

* INSERT: current behavior and default one.
* UPDATE: update documents instead of override. 
* insert_or_update: this is another useful action we can use and is main goal of this PR.

We should note somewhere that updates are partial.

makes sense?

@rhauch  do you agree with this?

Reply:
It sure does make sense to me.

My conclusion:
* we go with one property - currently named `write.method` (I am open to suggestions for new name if you want)
* we go with 3 acceptable values - INSERT, UPDATE, UPSERT (again, I am open to suggestions for different name if necessary)
* do proper documentation - where? is doc in class `ElasticsearchSinkConnectorConfig.java` sufficient? If yes, I will update doc description for that property.

Reply:
Class for config is ElasticsearchSinkConnectorConfig yes.

Reply:
Doc modified. Let me know whether it's OK.